subsys/net/l2/ethernet/gptp/gptp_messages.c

```diff
@@ -129,10 +129,25 @@ struct net_pkt *gptp_prepare_sync(int port)
 	struct net_buf *frag;
 	struct gptp_hdr *hdr;
 
+#if defined(CONFIG_NET_VLAN)
+	struct net_eth_vlan_hdr *hdr_vlan;
+	struct ethernet_context *eth_ctx;
+	bool vlan_enabled = false;
+#endif
+
 	NET_ASSERT((port >= GPTP_PORT_START) && (port <= GPTP_PORT_END));
 	iface = GPTP_PORT_IFACE(port);
 	NET_ASSERT(iface);
 
+#if defined(CONFIG_NET_VLAN)
+	eth_ctx = net_if_l2_data(iface);
```

Comment:
shouldn't you be using net_eth_is_vlan_enabled() here?

Btw, I can't recall why net_eth_is_vlan_enabled() has to take the context in parameter? Since we pass the iface, it can get the context. Or is this function meant to be used with a context that does not belong to the iface? (which sounds a bit weird)

Reply:
I will introduce a small helper to set the vlan header.